test/test_rohmu.py

```diff
@@ -0,0 +1,163 @@
+import hashlib
+import logging
+import os
+from tempfile import NamedTemporaryFile
+
+import pytest
+from rohmu import get_transfer, rohmufile
+from rohmu.rohmufile import create_sink_pipeline
+
+from .base import CONSTANT_TEST_RSA_PRIVATE_KEY, CONSTANT_TEST_RSA_PUBLIC_KEY
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "compress_algorithm, file_size",
+    [("lzma", 0), ("snappy", 0), ("zstd", 0), ("lzma", 1), ("snappy", 1), ("zstd", 1)],
+    ids=[
+        "test_lzma_0byte_file", "test_snappy_0byte_file", "test_zstd_0byte_file", "test_lzma_1byte_file",
+        "test_snappy_1byte_file", "test_zstd_1byte_file"
+    ],
+)
+def test_rohmu_with_local_storage(compress_algorithm: str, file_size: int, tmp_path):
+    hash_algorithm = "sha1"
+    compression_level = 0
+
+    # 0 - Prepare the file
+    work_dir = tmp_path
+    orig_file = work_dir / "hello.bin"
+    content = os.urandom(file_size)
+    with open(orig_file, "wb") as file_out:
+        file_out.write(content)
+
+    with open(orig_file, "rb") as file_in:
+        assert file_in.read() == content
+
+    # 1 - Compressed the file
+    compressed_filepath = work_dir / "compressed" / "hello_compressed"
+    compressed_filepath.parent.mkdir(exist_ok=True)
+    hasher = hashlib.new(hash_algorithm)
+    input_obj = open(orig_file, "rb")
+    output_obj = NamedTemporaryFile(
+        dir=os.path.dirname(compressed_filepath), prefix=os.path.basename(compressed_filepath), suffix=".tmp-compress"
+    )
+    with input_obj, output_obj:
+        original_file_size, compressed_file_size = rohmufile.write_file(
+            data_callback=hasher.update,
+            input_obj=input_obj,
+            output_obj=output_obj,
+            compression_algorithm=compress_algorithm,
+            compression_level=compression_level,
+            rsa_public_key=CONSTANT_TEST_RSA_PUBLIC_KEY,
+            log_func=log.debug,
+        )
+        os.link(output_obj.name, compressed_filepath)
```

Comment:
Not really. The Python temp file implementation has not implement the `hardlink_to` method.

Reply:
Oh, my bad. Ah now I see why it's `.name` and still works here.
Still, in order to avoid using lower level functions, this might still be considered:
```python
Path(output_obj.name).hardlink_to(compressed_filepath)
```

Reply:
This `pathlib` link API implementation is inconsistent. Python3.8, 3.9 have no implementation of `hardlink_to` (have `link_to` instead) , it would not work on those two versions.
 
Also, the internal implementation of `hardlink_to` on Python3.10 and Python3.11 (it uses directly `os.link`) are slightly different.

Also, notice that: `the order of arguments (self, target) is the reverse of os.link's.`

I will keep this unchanged at the moment.